src/main/java/com/melon/portfoliomanager/services/HttpService.java

```diff
@@ -0,0 +1,61 @@
+package com.melon.portfoliomanager.services;
+
+import com.melon.portfoliomanager.dtos.responses.StockPricesDTO;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Service;
+import org.springframework.web.reactive.function.client.WebClient;
+
+import java.net.URI;
+
+@Service
+public class HttpService {
+    private static final Logger logger = LoggerFactory.getLogger(HttpService.class);
+    private final WebClient webClient;
+
+    @Autowired
+    public HttpService(WebClient webClient) {
+        this.webClient = webClient;
+    }
+
+    public ResponseEntity<StockPricesDTO> getStockPrices(URI uri) {
+        ResponseEntity<StockPricesDTO> response = webClient.get()
```

Comment:
it may be worth to add a parameter to determine the http verb

Reply:
This is also something not needed regarding the idea behind the HttpService class. Please refer to my comment here: https://github.com/Hypermood/Portfolio-Manager/pull/8#discussion_r1287116022